src/python/gudhi/delaunay_complex.pyx

```diff
@@ -60,23 +62,21 @@ cdef class DelaunayComplex:
     cdef Delaunay_complex_interface * this_ptr
 
     # Fake constructor that does nothing but documenting the constructor
-    def __init__(self, points=[], weights=None, precision='safe'):
+    def __init__(self, points: Iterable[Iterable[float]] = [], weights: Optional[Iterable[float]] = None,
+                 precision: Literal['fast', 'safe', 'exact'] = 'safe'):
         """DelaunayComplex constructor.
 
-        :param points: A list of points in d-Dimension.
-        :type points: Iterable[Iterable[float]]
-
-        :param weights: A list of weights. If set, the number of weights must correspond to the number of points.
-        :type weights: Iterable[float]
-
-        :param precision: Delaunay complex precision can be 'fast', 'safe' or 'exact'. Default is 'safe'.
-        :type precision: string
+        Args:
+            points (Iterable[Iterable[float]]): A list of points in d-Dimension.
```

Comment:
Isn't the type redundant?

Reply:
It seems you have to declare it like that with mypy, for example:
```python
from collections.abc import Iterable

def dummy_sum(a: Iterable[Iterable[int]]):
    sum = 0
    for line in a:
        for value in line:
            sum += value
    return sum

arr = [[1,1],[1,5]]
print(f'sum = {dummy_sum(arr)}')
```

while, if I declare `dummy_sum` as `def dummy_sum(a: Iterable[int]):`, mypy is unhappy:
```txt
test_mypy.py:6: error: "int" has no attribute "__iter__"; maybe "__int__"? (not iterable)  [attr-defined]
test_mypy.py:11: error: Argument 1 to "dummy_sum" has incompatible type "list[list[int]]"; expected "Iterable[int]"  [arg-type]
Found 2 errors in 1 file (checked 1 source file)
```